### PR TITLE
Example: adding tests to verify the basic app

### PR DESCRIPTION
**Pull request by mckn**
This should probably be good enough to verify that this app is working as expected. I renamed the test specs in the datasource to be consistent with the rest of the spec files in the repo.

Fixes: #53 #12

### CODE REVIEW

File: examples/app-basic/src/plugin.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "$schema": "https://raw.githubusercontent.com/grafana/grafana/master/docs/sources/developers/plugins/plugin.schema.json",
   "type": "app",
   "name": "Basic App",
-  "id": "basic-app",
+  "id": "grafana-basic-app",
```
Comment (mckn) [inline examples/app-basic/src/plugin.json:5]: Prefixing with `grafana-` to remove risk of colliding.

File: examples/app-basic/cypress/integration/01-app-config.spec.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { e2e } from '@grafana/e2e';
+import { testIds } from '../../src/components/testIds';
+
+const { appConfig } = e2e.getSelectors(testIds);
+
+describe('configurating app', () => {
+  beforeEach(() => {
+    cy.visit('http://localhost:3000/plugins/grafana-basic-app');
```
Comment (jackw) [inline examples/app-basic/cypress/integration/01-app-config.spec.ts]: We should be able to improve this with a flow to e2e package or some such.

**Review by academo — APPROVED** [review]
Nice! thanks for adding those tests.

File: examples/app-basic/cypress/integration/02-app-navigation.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  });
+
+  it('should render full width example successfully', () => {
+    // wait for page to successfully render
+    pageOne.container().should('be.visible');
+    cy.get('h1').contains(headerTitle).should('be.visible');
+    
+    // navigating to page four with full width layout
+    pageOne.navigateToFour().click();
+    cy.get('body').find('h1', ).should('not.exist');
```
Comment (jackw) [inline examples/app-basic/cypress/integration/02-app-navigation.spec.ts:26]: Is this syntax correct?

```suggestion
    cy.get('body').find('h1').should('not.exist');
```

File: examples/app-basic/src/plugin.json (hunk 1 above)
Comment (jackw) [reply examples/app-basic/src/plugin.json]: Isn't the prefix of a plugin id be the org name? Maybe swap it out for `myOrg`?

`"id": "myorg-basic-app",`

**Review by jackw — APPROVED** [review]
Awesome work @mckn! 🚀 

Left a couple of nits.

File: examples/app-basic/cypress/integration/01-app-config.spec.ts (hunk 2 above)
Comment (mckn) [reply examples/app-basic/cypress/integration/01-app-config.spec.ts]: I totally agree! Lets add it to the list to expose some kind of function that will fetch this based on the environment we are running the tests in.

File: examples/app-basic/cypress/integration/02-app-navigation.spec.ts (hunk 3 above)
Comment (mckn) [reply examples/app-basic/cypress/integration/02-app-navigation.spec.ts:26]: Oops, my bad 🙈  Thanks for the ping.

**Pull request merged**